src/rpc/core_rpc_server.cpp

```diff
@@ -690,6 +695,13 @@ namespace cryptonote
         e.block_timestamp = m_core.get_blockchain_storage().get_db().get_block_timestamp(e.block_height);
         e.double_spend_seen = false;
         e.relayed = false;
+        if (e.block_height > immutable_height)
+        {
+          if (!pool_lock) pool_lock.lock();
+          e.blink = pool.has_blink(tx_hash, true /*have lock*/);
```

Comment:
`e.blink = pool.has_blink(tx_hash, false /*have lock*/);`

To avoid double locking UB.

Reply:
I want the scope of the lock to be over the whole loop, but only if I need it since this function gets used for lots of older immutable txs, too, so often I won't need blink querying at all.  Thus the lock is initially deferred and only actually acquired here (once) in the previous line.

Reply:
oh i see. Read have_lock as, take the lock. All good.